In [ ]:
# access file from Google Drive

from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
import pandas as pd

In [ ]:
file_path = '/content/drive/My Drive/ESIIL Project Data/Population Data/Pop Data - Douglas_Eagle_Logan.csv'
df_new = pd.read_csv(file_path)

display(df_new.head())
display(df_new.info())

,Year,Douglas Population,Eagle Population,Logan Population
0,1985,"40,048","18,029","19,063"
1,1986,"45,654","18,853","18,167"
2,1987,"48,844","19,002","17,740"
3,1988,"53,127","19,567","17,932"
4,1989,"57,164","20,881","17,725"


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 38 entries, 0 to 37
Data columns (total 4 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   Year                38 non-null     int64 
 1   Douglas Population  38 non-null     object
 2   Eagle Population    38 non-null     object
 3   Logan Population    38 non-null     object
dtypes: int64(1), object(3)
memory usage: 1.3+ KB


None

## Data preparation

### Subtask:
Reshape the data to be suitable for ANOVA, which typically requires a long format with columns for population, county, and year.

**Reasoning**:
Clean the population columns and reshape the DataFrame to a long format suitable for ANOVA.

In [ ]:
for col in ['Douglas Population', 'Eagle Population', 'Logan Population']:
    df_new[col] = df_new[col].str.replace(',', '').astype(int)

df_melted_new = df_new.melt(id_vars=['Year'],
                            value_vars=['Douglas Population', 'Eagle Population', 'Logan Population'],
                            var_name='County',
                            value_name='Population')

df_melted_new['County'] = df_melted_new['County'].str.replace(' Population', '')

display(df_melted_new.head())

,Year,County,Population
0,1985,Douglas,40048
1,1986,Douglas,45654
2,1987,Douglas,48844
3,1988,Douglas,53127
4,1989,Douglas,57164


In [ ]:
print(df_melted_new)

     Year   County  Population
0    1985  Douglas       40048
1    1986  Douglas       45654
2    1987  Douglas       48844
3    1988  Douglas       53127
4    1989  Douglas       57164
5    1990  Douglas       61559
6    1991  Douglas       66688
7    1992  Douglas       75361
8    1993  Douglas       83132
9    1994  Douglas       92526
10   1995  Douglas      103839
11   1996  Douglas      114713
12   1997  Douglas      129331
13   1998  Douglas      144354
14   1999  Douglas      162323
15   2000  Douglas      180510
16   2001  Douglas      199038
17   2002  Douglas      209705
18   2003  Douglas      221146
19   2004  Douglas      233646
20   2005  Douglas      244442
21   2006  Douglas      257833
22   2007  Douglas      268599
23   2008  Douglas      276740
24   2009  Douglas      282163
25   2010  Douglas      287124
26   2011  Douglas      292623
27   2012  Douglas      298733
28   2013  Douglas      306327
29   2014  Douglas      314609
30   2015  Douglas      322079
31   201

## Anova analysis (1985-2001)

### Subtask:
Perform ANOVA on the data for the years 1985 to 2001.

**Reasoning**:
Filter the dataframe for the specified years and perform the ANOVA analysis.

In [ ]:
df_1985_2001_new = df_melted_new[(df_melted_new['Year'] >= 1985) & (df_melted_new['Year'] <= 2001)]

from statsmodels.formula.api import ols
from statsmodels.stats.anova import anova_lm

model_1985_2001_new = ols('Population ~ C(County)', data=df_1985_2001_new).fit()
anova_table_1985_2001_new = anova_lm(model_1985_2001_new)

print(anova_table_1985_2001_new)

             df        sum_sq       mean_sq          F        PR(>F)
C(County)   2.0  6.242266e+10  3.121133e+10  36.396547  2.403140e-10
Residual   48.0  4.116170e+10  8.575355e+08        NaN           NaN


## Anova analysis (2002-2022)

### Subtask:
Perform ANOVA on the data for the years 2002 to 2022.

**Reasoning**:
Filter the data for the years 2002-2022, fit an OLS model, perform ANOVA, and print the results.

In [ ]:
df_2002_2022_new = df_melted_new[(df_melted_new['Year'] >= 2002) & (df_melted_new['Year'] <= 2022)]

model_2002_2022_new = ols('Population ~ C(County)', data=df_2002_2022_new).fit()
anova_table_2002_2022_new = anova_lm(model_2002_2022_new)

print(anova_table_2002_2022_new)

             df        sum_sq       mean_sq           F        PR(>F)
C(County)   2.0  9.730607e+11  4.865303e+11  614.417662  1.092986e-40
Residual   60.0  4.751136e+10  7.918560e+08         NaN           NaN


## Anova analysis (1985-2022)

### Subtask:
Perform ANOVA on the data for all years 1985 to 2022.

**Reasoning**:
Perform ANOVA on the entire dataset (1985-2022) using the previously melted and cleaned DataFrame `df_melted_new`.

In [ ]:
model_1985_2022_new = ols('Population ~ C(County)', data=df_melted_new).fit()
anova_table_1985_2022_new = anova_lm(model_1985_2022_new)

print(anova_table_1985_2022_new)

              df        sum_sq       mean_sq          F        PR(>F)
C(County)    2.0  8.107257e+11  4.053629e+11  94.685753  1.012431e-24
Residual   111.0  4.752064e+11  4.281139e+09        NaN           NaN


## Present results

### Subtask:
Display the results of the ANOVA analyses.

**Reasoning**:
Display the results of the three ANOVA analyses by printing the respective ANOVA tables.

In [ ]:
print("ANOVA Table for 1985-2001:")
print(anova_table_1985_2001_new)
print("\nANOVA Table for 2002-2022:")
print(anova_table_2002_2022_new)
print("\nANOVA Table for 1985-2022:")
print(anova_table_1985_2022_new)

ANOVA Table for 1985-2001:
             df        sum_sq       mean_sq          F        PR(>F)
C(County)   2.0  6.242266e+10  3.121133e+10  36.396547  2.403140e-10
Residual   48.0  4.116170e+10  8.575355e+08        NaN           NaN

ANOVA Table for 2002-2022:
             df        sum_sq       mean_sq           F        PR(>F)
C(County)   2.0  9.730607e+11  4.865303e+11  614.417662  1.092986e-40
Residual   60.0  4.751136e+10  7.918560e+08         NaN           NaN

ANOVA Table for 1985-2022:
              df        sum_sq       mean_sq          F        PR(>F)
C(County)    2.0  8.107257e+11  4.053629e+11  94.685753  1.012431e-24
Residual   111.0  4.752064e+11  4.281139e+09        NaN           NaN


## Export ANOVA Results to CSV

### Subtask:
Export the ANOVA tables for each time period to separate CSV files.

**Reasoning**:
Use the `to_csv()` method to save each ANOVA table DataFrame to a CSV file with an appropriate filename.

In [ ]:
# Export ANOVA table for 1985-2001
anova_table_1985_2001_new.to_csv('/content/drive/My Drive/ANOVA_Douglas_Eagle_Logan_1985_2001.csv')

# Export ANOVA table for 2002-2022
anova_table_2002_2022_new.to_csv('/content/drive/My Drive/ANOVA_Douglas_Eagle_Logan_2002_2022.csv')

# Export ANOVA table for 1985-2022
anova_table_1985_2022_new.to_csv('/content/drive/My Drive/ANOVA_Douglas_Eagle_Logan_1985_2022.csv')

print("ANOVA results for Douglas, Eagle, and Logan counties exported to CSV files in your Google Drive.")

ANOVA results for Douglas, Eagle, and Logan counties exported to CSV files in your Google Drive.


## Summary:

### Data Analysis Key Findings

* The data for Douglas, Eagle, and Logan counties was successfully loaded, cleaned, and reshaped for ANOVA.
* For the period 1985-2001, the ANOVA analysis showed a statistically significant difference in the mean population between Douglas, Eagle, and Logan counties (F-statistic = 36.40, p-value = 2.40e-10).
* For the period 2002-2022, the ANOVA analysis also showed a statistically significant difference in the mean population between the three counties (F-statistic = 614.42, p-value = 1.09e-40).
* For the entire period 1985-2022, the ANOVA analysis indicated a statistically significant difference in the mean population between Douglas, Eagle, and Logan counties (F-statistic = 94.69, p-value = 1.01e-24).
* The ANOVA results for each time period have been exported to separate CSV files in your Google Drive.

### Insights or Next Steps

* Similar to the previous analysis, post-hoc tests (e.g., Tukey's HSD) could be conducted to identify which specific pairs of counties have significantly different mean populations within each time period.
* Visualizing the population trends for each of these counties over time would provide valuable context for the ANOVA results and help understand the population dynamics in each county.
* Comparing the ANOVA results and trends of these three counties with the previously analyzed counties (Baca, Douglas, and Eagle) could offer interesting insights into regional population changes.